hazelcast/src/main/java/com/hazelcast/jet/config/JobConfigArguments.java

```diff
@@ -34,5 +34,11 @@ public final class JobConfigArguments {
      */
     public static final String KEY_SQL_UNBOUNDED = "__sql.queryUnbounded";
 
+    /**
+     * The key under which the SQL engine stores detected partitions to apply
+     * member pruning technique.
+     */
+    public static final String KEY_REQUIRED_PARTITIONS = "__sql.requiredPartitions";
```

Comment:
> Why is this argument SQL only?

Because we designed partition pruning to be SQL-specific. SQL optimizer is an additional analysis phase that is able to apply member pruning as optimization.

Reply:
To know more : https://github.com/hazelcast/hazelcast/pull/24596